go/mcap/writer.go

```diff
@@ -284,7 +284,7 @@ func (w *Writer) WriteAttachmentIndex(idx *AttachmentIndex) error {
 	if w.opts.SkipAttachmentIndex {
 		return nil
 	}
-	msglen := 8 + 8 + 8 + 8 + 4 + len(idx.Name) + 4 + len(idx.ContentType)
+	msglen := 8 + 8 + 8 + 8 + 8 + 4 + len(idx.Name) + 4 + len(idx.ContentType)
```

Comment:
what caused this change?

Reply:
oh I see - it was just wrong. probably never exercised... :(

Reply:
I tried to explain it in the commit msg: https://github.com/foxglove/mcap/pull/445/commits/a518ee38568cb18c05eebe9dde7c893ace353cde

basically it's missing 8 bytes for one of the uint64s. Previously we didn't notice because `putStringPrefixed` silently truncates when there's not enough space, and all of out attachment content type strings were more than 8 characters.